runners/direct-java/src/test/java/org/apache/beam/runners/direct/DirectRunnerTest.java

```diff
@@ -131,13 +141,23 @@ public String apply(KV<String, Long> input) {
           }
         }));
 
+    counts.apply(ParDo.of(new DoFn<KV<String, Long>, Void>() {
+      @ProcessElement
+      public void updateChanged(ProcessContext c) {
+        changed.getAndIncrement();
+      }
+    }));
+
+
     PAssert.that(countStrs).containsInAnyOrder("baz: 1", "bar: 2", "foo: 3");
 
     DirectPipelineResult result = ((DirectPipelineResult) p.run());
     result.awaitCompletion();
 
     DirectPipelineResult otherResult = ((DirectPipelineResult) p.run());
     otherResult.awaitCompletion();
+
+    assertThat(changed.get(), equalTo(6));
```

Comment:
`assertThat("Make sure that each element was processed twice", changed.get(), equalTo(6));`